sdk/eventgrid/eventgrid/review/eventgrid.api.md

```diff
@@ -99,47 +131,121 @@ export type ContainerRegistryImageDeletedEventData = ContainerRegistryEventData
 export type ContainerRegistryImagePushedEventData = ContainerRegistryEventData & {};
 
 // @public
-export class EventGridClient {
-    constructor(endpointUrl: string, credential: KeyCredential | SignatureCredential, options?: EventGridClientOptions);
-    readonly apiVersion: string;
-    readonly endpointUrl: string;
-    generateSharedAccessSignature(expiresOnUtc: Date): string;
-    sendCloudEvents(events: CloudEvent<any>[], options?: SendCloudEventsOptions): Promise<RestResponse>;
-    sendCustomSchemaEvents(events: Record<string, any>[], options?: SendCustomSchemaEventsOptions): Promise<RestResponse>;
-    sendEvents(events: EventGridEvent<any>[], options?: SendEventsOptions): Promise<RestResponse>;
+export type CustomEventDataDeserializer = (o: any) => Promise<any>;
```

Comment:
is it worth making this

```suggestion
export type CustomEventDataDeserializer<Input, Output> = (o:  Input) => Promise<Output>;
```

Reply:
I'm not sure.  In practice it would mean that `EventGridConsumerOptions` would take a `CustomEventDataDeserializer<any, any>[]` array, since we'd have to specify some type parameters there, so I think at the end of the day I'm not sure it would help much?

Reply:
Yeah if it's not a type we can easily flow down, I think it's fine to leave it at `any`